src/systems/filecoin_blockchain/struct/block_producer/_index.md

```diff
@@ -46,73 +49,64 @@ Anytime a miner receives new valid blocks, it should evaluate what is the heavie
-    - `BLSMessages`: the root cid of the bls messages sharray
-    - `SECPMessages`: the root cid of the secp messages sharray
-  - The cid of this `TxMeta` object should be used to fill the `Messages` field of the block header.
-- `BLSAggregate` - The aggregated signatures of all messages in the block that used BLS signing.
-- `Signature` - A signature with the miner's worker account private key (must also match the ticket signature) over the entire block. This is to ensure that nobody tampers with the block after it propagates to the network, since unlike normal PoW blockchains, a winning ticket is found independently of block generation.
-
-An eligible miner can start by filling out `Parents`, `Tickets` and `ElectionPoStVerifyInfo`.
-
-Next, they compute the aggregate state of their selected parent blocks, the `ParentState`. This is done by taking the aggregate parent state of the blocks' parent Tipset, sorting the parent blocks by their tickets, and applying each message in each block to that state. Any message whose nonce is already used (duplicate message) in an earlier block should be skipped (application of this message should fail anyway). Note that re-applied messages may result in different receipts than they produced in their original blocks, an open question is how to represent the receipt trie of this tipset's messages (one can think of a tipset as a 'virtual block' of sorts).
+  - `TxMeta.BLSMessages`: The CID of the root of an AMT comprising the bare `UnsignedMessage`s
+  - `TxMeta.SECPMessages`: the CID of the root of an AMT comprising the `SignedMessage`s
+- `BLSAggregate` - The aggregated signature of all messages in the block that used BLS signing.
+- `Signature` - A signature with the miner's worker account private key (must also match the ticket signature) over the the block header's serialized representation (with empty signature). 
 
-They gather the receipts from each above message execution into a set, merklize them, and put that root in `ReceiptsRoot`. 
+Note that the messages to be included in a block need not be evaluated in order to produce a valid block.
+A miner may wish to speculatively evaluate the messages anyway in order to optimize for including
+messages which will succeed in execution and pay the most gas.
 
-Once the miner has the aggregate `ParentState`, they must apply the block reward. This is done by adding the correct block reward amount to the miner owner's account balance in the state tree. The reward will be spendable immediately in this block.
+The block reward is not evaluated when producing a block. It is paid when the block is included in a tipset in the following epoch.
 
-Now, a set of messages is selected to put into the block. For each message, the miner subtracts `msg.GasPrice * msg.GasLimit` from the sender's account balance, returning a fatal processing error if the sender does not have enough funds (this message should not be included in the chain).
```

Comment:
Is this info included elsewhere in md?

Reply:
No, it's wrong. The block producer need not do any of this.